core/src/main/scala/com/salesforce/op/stages/impl/feature/HumanNameDetector.scala

```diff
@@ -0,0 +1,162 @@
+ * for English at the moment). If the column does not appear to be a name, then the output will be an empty map.
+ * @param uid           uid for instance
+ * @param operationName unique name of the operation this stage performs
+ * @param tti           type tag for input
+ * @param ttiv          type tag for input value
+ * @tparam T            the FeatureType (subtype of Text) to operate over
+ */
+class HumanNameDetector[T <: Text]
+(
+  uid: String = UID[HumanNameDetector[T]],
+  operationName: String = "human name detector"
+)
+(
+  implicit tti: TypeTag[T],
+  override val ttiv: TypeTag[T#Value]
+) extends UnaryEstimator[T, NameStats](
+  uid = uid,
+  operationName = operationName
+) with NameDetectFun {
+
+  def fitFn(dataset: Dataset[T#Value]): HumanNameDetectorModel[T] = {
+    implicit val (nameDetectStatsEnc, nameDetectStatsMonoid) = (NameDetectStats.kryo, NameDetectStats.monoid)
+    val mapFun: T#Value => NameDetectStats = makeMapFunction(dataset.sparkSession)
+    val aggResults: NameDetectStats = dataset.map(mapFun).reduce(_ + _)
+    val treatAsName = computeTreatAsName(aggResults)
+
+    val orderedGenderDetectStrategies = if (treatAsName) {
+      val ordered: Seq[(String, GenderStats)] = aggResults.genderResultsByStrategy.toSeq.sortBy(_._2.numOther)
+      ordered map { case (strategy, _) => GenderDetectStrategy.fromString(strategy) }
+    } else Seq.empty[GenderDetectStrategy]
+
+    val newMetadata = HumanNameDetectorMetadata(
+      treatAsName, aggResults.dictCheckResult.value, aggResults.genderResultsByStrategy
+    ).toMetadata()
+    val metaDataBuilder = new MetadataBuilder().withMetadata(getMetadata()).withMetadata(newMetadata)
+    setMetadata(metaDataBuilder.build())
+
+    new HumanNameDetectorModel[T](uid, treatAsName, orderedGenderDetectStrategies)
+  }
+}
+
+class HumanNameDetectorModel[T <: Text]
+(
+  override val uid: String,
+  val treatAsName: Boolean,
+  val orderedGenderDetectStrategies: Seq[GenderDetectStrategy] = Seq.empty[GenderDetectStrategy]
+)(implicit tti: TypeTag[T])
+  extends UnaryModel[T, NameStats]("human name detector", uid) with NameDetectFun {
+
+  import NameStats.BooleanStrings._
+  import NameStats.GenderStrings.GenderNA
+  import NameStats.Keys._
+  def transformFn: T => NameStats = (input: T) => {
+    val tokens = preProcess(input.value)
+    if (treatAsName) {
+      require(orderedGenderDetectStrategies.nonEmpty, "There must be a gender extraction strategy if treating as name.")
+      // TODO: Figure out how to use a broadcast variable for the gender dictionary within a unary transformer
+      val genders: Seq[String] = orderedGenderDetectStrategies map {
+        identifyGender(input.value.getOrElse(""), tokens, _, NameDetectUtils.DefaultGenderDictionary)
+      }
+      val gender = genders.find(_ != GenderNA).getOrElse(GenderNA)
+      NameStats(Map(
+        IsNameIndicator -> True,
```

Comment:
`IsNameIndicator -> treatAsName`

Reply:
Is that clearer? This bit of code is in the conditional branch when `treatAsName` already.